llvm/include/llvm/Analysis/TargetTransformInfo.h

```diff
@@ -665,6 +665,8 @@ class TargetTransformInfo {
   void getPeelingPreferences(Loop *L, ScalarEvolution &SE,
                              PeelingPreferences &PP) const;
 
+  bool isOffsetFoldingLegal(const GlobalValue *GV) const;
```

Comment:
This needs documentation. I've never had any idea what isOffsetFoldingLegal means

Reply:
done. PTAL, thanks!

Reply:
This still doesn't explain to me what "legal" means in this context. What offset? And for what value?